### PR TITLE
[pipeline](refactor) remove pipeline task state

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

File: be/src/pipeline/task_scheduler.cpp [hunk 1]
```diff
@@ -86,12 +86,10 @@ void _close_task(PipelineTask* task, PipelineTaskState state, Status exec_status
     // We have already refactor all source and sink api, the close API does not need waiting
     // for pending finish now. So that could call close directly.
     Status status = task->close(exec_status);
-    if (!status.ok() && state != PipelineTaskState::CANCELED) {
+    if (!status.ok() && !cancel) {
```
Comment (yiguolei) [inline be/src/pipeline/task_scheduler.cpp:89]: do not need this flag, the cancel logic should be 幂等

File: be/src/pipeline/task_scheduler.cpp [hunk 2]
```diff
@@ -113,25 +111,18 @@ void TaskScheduler::_do_work(size_t index) {
         auto* fragment_ctx = task->fragment_context();
         bool canceled = fragment_ctx->is_canceled();
 
-        auto state = task->get_state();
         // If the state is PENDING_FINISH, then the task is come from blocked queue, its is_pending_finish
         // has to return false. The task is finished and need to close now.
-        if (state == PipelineTaskState::PENDING_FINISH || canceled) {
```
Comment (yiguolei) [inline be/src/pipeline/task_scheduler.cpp:119]: 这个逻辑不一定对。
当finishdpendency 中唤醒的时候，需要在这里判断一下，是否直接close了

File: be/src/pipeline/pipeline_task.cpp [hunk 3]
```diff
@@ -205,6 +201,10 @@ Status PipelineTask::execute(bool* eos) {
     SCOPED_TIMER(_task_profile->total_time_counter());
     SCOPED_TIMER(_exec_timer);
     SCOPED_ATTACH_TASK(_state);
+    *eos = _eos;
```
Comment (yiguolei) [inline be/src/pipeline/pipeline_task.cpp:204]: add comment here

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by Mryange — APPROVED** [review]

**Pull request merged**